### PR TITLE
Issue 112599 - fix pydocstyle errors

### LINKED ISSUE #112599: Fix docstring errors in pixelshuffle.py, sparse.py, rnn.py, pooling.py, upsampling.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelShuffle`, **Line**: 9, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelShuffle`, **Line**: 9, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelShuffle`, **Line**: 9, **Description**: First line should end with a period (not '`')
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelUnshuffle`, **Line**: 61, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelUnshuffle`, **Line**: 61, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pixelshuffle.py`, **Entity**: `PixelUnshuffle`, **Line**: 61, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool1d`, **Line**: 39, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool1d`, **Line**: 39, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool2d`, **Line**: 98, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool2d`, **Line**: 98, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool3d`, **Line**: 172, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `MaxPool3d`, **Line**: 172, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool1d`, **Line**: 498, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool1d`, **Line**: 498, **Description**: First line should end with a period (not 'l')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool2d`, **Line**: 565, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool2d`, **Line**: 565, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool2d`, **Line**: 565, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool3d`, **Line**: 644, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool3d`, **Line**: 644, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `AvgPool3d`, **Line**: 644, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `FractionalMaxPool2d`, **Line**: 736, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `FractionalMaxPool3d`, **Line**: 806, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `LPPool1d`, **Line**: 894, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `LPPool1d`, **Line**: 894, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `LPPool2d`, **Line**: 936, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/pooling.py`, **Entity**: `LPPool2d`, **Line**: 936, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `Embedding`, **Line**: 14, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `from_pretrained`, **Line**: 184, **Description**: First line should be in imperative mood (perhaps 'Create', not 'Creates')
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `EmbeddingBag`, **Line**: 227, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `EmbeddingBag`, **Line**: 227, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `EmbeddingBag`, **Line**: 227, **Description**: First line should end with a period (not 'e')
- **File**: `torch/nn/modules/sparse.py`, **Entity**: `from_pretrained`, **Line**: 413, **Description**: First line should be in imperative mood (perhaps 'Create', not 'Creates')
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `flatten_parameters`, **Line**: 149, **Description**: First line should be in imperative mood (perhaps 'Reset', not 'Resets')
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `RNN`, **Line**: 345, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `RNN`, **Line**: 345, **Description**: First line should end with a period (not 'n')
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `LSTM`, **Line**: 565, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `LSTM`, **Line**: 565, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `GRU`, **Line**: 852, **Description**: 1 blank line required between summary line and description (found 2)
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `RNNCell`, **Line**: 1110, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/rnn.py`, **Entity**: `GRUCell`, **Line**: 1297, **Description**: First line should end with a period (not 'l')
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `Upsample`, **Line**: 12, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingNearest2d`, **Line**: 175, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingNearest2d`, **Line**: 175, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingNearest2d`, **Line**: 175, **Description**: First line should end with a period (not 't')
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingBilinear2d`, **Line**: 220, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingBilinear2d`, **Line**: 220, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/upsampling.py`, **Entity**: `UpsamplingBilinear2d`, **Line**: 220, **Description**: First line should end with a period (not 't')

cc @carljparker

### PR DESCRIPTION
**Pull request by zabboud**
Fixes #112599 

Fixed errors relating to pydocstyle in the following files. The remaining errors are related to docstrings at the module level and at methods within each module, `forward()`, `reset_parameters`, `__init__` ..etc 

pydocstyle torch/nn/modules/pooling.py --count
before: 49
after: 29 

**remaining errors:**
```
torch/nn/modules/pooling.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/modules/pooling.py:90 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:163 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:240 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:315 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:321 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:402 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:408 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:472 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:478 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:541 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:550 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:620 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:630 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:706 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:716 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:720 in public method `__setstate__`:
        D105: Missing docstring in magic method
torch/nn/modules/pooling.py:774 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:792 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:845 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pooling.py:863 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:925 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:979 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1026 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1068 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1111 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1150 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1189 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pooling.py:1228 in public method `forward`:
        D102: Missing docstring in public method
```

pydocstyle torch/nn/modules/upsampling.py --count
before: 14
after: 7

**remaining:**
```
torch/nn/modules/upsampling.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/modules/upsampling.py:142 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/upsampling.py:156 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/upsampling.py:160 in public method `__setstate__`:
        D105: Missing docstring in magic method
torch/nn/modules/upsampling.py:166 in public method `extra_repr`:
        D102: Missing docstring in public method
torch/nn/modules/upsampling.py:216 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/upsampling.py:263 in public method `__init__`:
        D107: Missing docstring in __init__
```

pydocstyle torch/nn/modules/rnn.py --count
before: 47
after: 40

**remaining**
```
torch/nn/modules/rnn.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/modules/rnn.py:59 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:160 in public method `__setattr__`:
        D105: Missing docstring in magic method
torch/nn/modules/rnn.py:225 in public method `reset_parameters`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:230 in public method `check_input`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:242 in public method `get_expected_hidden_size`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:256 in public method `check_hidden_size`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:272 in public method `check_forward_args`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:278 in public method `permute_hidden`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:284 in public method `extra_repr`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:305 in public method `__getstate__`:
        D105: Missing docstring in magic method
torch/nn/modules/rnn.py:313 in public method `__setstate__`:
        D105: Missing docstring in magic method
torch/nn/modules/rnn.py:355 in public method `all_weights`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:471 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:478 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:481 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:503 in public method `forward` (skipping F811):
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:762 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:768 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:771 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:774 in public method `get_expected_cell_size`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:786 in public method `check_forward_args`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:798 in public method `permute_hidden`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:809 in public method `forward` (skipping F811):
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:820 in public method `forward` (skipping F811):
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1030 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1036 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1039 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1046 in public method `forward` (skipping F811):
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1054 in public method `forward` (skipping F811):
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1123 in public class `RNNCellBase`:
        D101: Missing docstring in public class
torch/nn/modules/rnn.py:1134 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1152 in public method `extra_repr`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1160 in public method `reset_parameters`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1224 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1230 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1327 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1332 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/rnn.py:1422 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/rnn.py:1427 in public method `forward`:
        D102: Missing docstring in public method
```

pydocstyle torch/nn/modules/pixelshuffle.py --count
before: 13
after: 8

**remaining:**
```
torch/nn/modules/pixelshuffle.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/modules/pixelshuffle.py:52 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pixelshuffle.py:56 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pixelshuffle.py:59 in public method `extra_repr`:
        D102: Missing docstring in public method
torch/nn/modules/pixelshuffle.py:105 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/pixelshuffle.py:109 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/pixelshuffle.py:112 in public method `extra_repr`:
        D102: Missing docstring in public method
```

pydocstyle torch/nn/modules/sparse.py --count
before: 14
after: 8

**remaining errors:**
```
torch/nn/modules/sparse.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/modules/sparse.py:124 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/sparse.py:153 in public method `reset_parameters`:
        D102: Missing docstring in public method
torch/nn/modules/sparse.py:162 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/modules/sparse.py:167 in public method `extra_repr`:
        D102: Missing docstring in public method
torch/nn/modules/sparse.py:320 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/modules/sparse.py:350 in public method `reset_parameters`:
        D102: Missing docstring in public method
torch/nn/modules/sparse.py:396 in public method `extra_repr`:
        D102: Missing docstring in public method
```

### CODE REVIEW

**Review by mikaylagawarecki — COMMENTED** [review]
Hey @zabboud , could you please update the PR description with the errors that were not fixed / is it it possible to fix them in this PR as well?

**Comment by zabboud** [comment]
@mikaylagawarecki I've updated the PR description with the remaining errors - they include requirements of docstrings at the module level and mostly methods like `forward()` `__init__` `reset_parameters`, which probably don't fit with the general style of pytorch docstring documentation. The only one that I noticed that may have been an oversight would be the class `RNNCellBase` in `rnn.py`, which is missing a docstring. I could fix that if needed (it wasn't covered in the initial issue but I'd be happy to add a docstring to that class)

File: torch/nn/modules/rnn.py [hunk 1]
```diff
@@ -364,10 +365,9 @@ def _replicate_for_data_parallel(self):
 
 
 class RNN(RNNBase):
-    r"""__init__(self,input_size,hidden_size,num_layers=1,nonlinearity='tanh',bias=True,batch_first=False,dropout=0.0,bidirectional=False,device=None,dtype=None)
-
-    Applies a multi-layer Elman RNN with :math:`\tanh` or :math:`\text{ReLU}` non-linearity to an
-    input sequence.
+    r"""Apply a multi-layer Elman RNN with :math:`\tanh` or :math:`\text{ReLU}` non-linearity to an input sequence.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:368]: The signature on the first line is here intentionally as otherwise the sphinx documentation will not render the arguments properly, similarly for LSTM below.

It is okay to ignore the pydocstyle errors for these two and revert these change

You can double check this at the [docs-preview](https://docs-preview.pytorch.org/pytorch/pytorch/113177/index.html) when it renders

File: torch/nn/modules/rnn.py [hunk 2]
```diff
@@ -597,10 +597,9 @@ def forward(self, input, hx=None):  # noqa: F811
 
 
 class LSTM(RNNBase):
-    r"""__init__(self,input_size,hidden_size,num_layers=1,bias=True,batch_first=False,dropout=0.0,bidirectional=False,proj_size=0,device=None,dtype=None)
+    r"""Apply a multi-layer long short-term memory (LSTM) RNN to an input sequence.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:600]: Same comment as above

Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:600]: Same comment as above

File: torch/nn/modules/rnn.py (hunk 1 above)
Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:368]: The signature on the first line is here intentionally as otherwise the sphinx documentation will not render the arguments properly, similarly for LSTM and GRU below.

It is okay to ignore the pydocstyle errors for these two and revert these change

You can double check this at the [docs-preview](https://docs-preview.pytorch.org/pytorch/pytorch/113177/index.html) when it renders. Apologies for the churn

File: torch/nn/modules/rnn.py [hunk 3]
```diff
@@ -897,9 +896,9 @@ def forward(self, input, hx=None):  # noqa: F811
 
 
 class GRU(RNNBase):
-    r"""__init__(self,input_size,hidden_size,num_layers=1,bias=True,batch_first=False,dropout=0.0,bidirectional=False,device=None,dtype=None)
-
-    Applies a multi-layer gated recurrent unit (GRU) RNN to an input sequence.
+    r"""Apply a multi-layer gated recurrent unit (GRU) RNN to an input sequence.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/rnn.py:899]: Same comment as above

Comment (zabboud) [reply torch/nn/modules/rnn.py]: addressed in the latest commit - please confirm if it's okay.

File: torch/nn/modules/rnn.py (hunk 1 above)
Comment (zabboud) [reply torch/nn/modules/rnn.py]: addressed in the latest commit - please confirm if it's okay.

**Comment by zabboud** [comment]
@mikaylagawarecki  I've fixed the issues you've pointed out. could you please review them and let me know if anything else is needed?

**Review by mikaylagawarecki — COMMENTED** [review]
Please fix lint, otherwise this looks ok.

**Comment by zabboud** [comment]
@mikaylagawarecki it is just awaiting approval - all checks have passed.

**Comment by zabboud** [comment]
@mikaylagawarecki @svekars Can a merge be initiated to close the PR please?

**Review by ezyang — APPROVED** [review]
